File: Questão14.go
package main

import (
    "fmt"
    sync "sync"
)

type Configuracao struct {
    config map[string]string
}

var instancia *Configuracao
var once sync.Once


func GetInstancia() *Configuracao {
    once.Do(func() {
        instancia = &Configuracao{config: make(map[string]string)}
    })
    return instancia
}

func (c *Configuracao) SetConfig(chave string, valor string) {
    c.config[chave] = valor
}

func (c *Configuracao) GetConfig(chave string) string {
    if valor, existe := c.config[chave]; existe {
        return valor
    }
    return "Configuração não encontrada"
}

func main() {
    config1 := GetInstancia()
    config1.SetConfig("url", "http://exemplo.com")

    config2 := GetInstancia()
    fmt.Println(config2.GetConfig("url")) 
    fmt.Println(config1 == config2)        
}

File: Questão14.java
class Configuracao {
    private static Configuracao instancia;

    private Configuracao() {
       
    }

    public static Configuracao getInstancia() {
        if (instancia == null) {
            instancia = new Configuracao();
        }
        return instancia;
    }

   
    public String getConfiguracao(String chave) {
        
        return "Configuração não encontrada";
    }
}

public class Main {
    public static void main(String[] args) {
        Configuracao config1 = Configuracao.getInstancia();
        Configuracao config2 = Configuracao.getInstancia();
        System.out.println(config1 == config2); 
    }
}

